server/bin/pbench-server-prep-shim-002.py

```diff
@@ -311,51 +158,24 @@ def main(cfg_name):
 
     logger = get_pbench_logger(_NAME_, config)
 
-    # We're going to need the Postgres DB to track dataset state, so setup
-    # DB access.
-    init_db(config, logger)
-
-    qdir, receive_dir = fetch_config_val(config, logger)
+    receive_dir = fetch_config_val(config, logger)
 
-    if qdir is None and receive_dir is None:
+    if receive_dir is None:
         return 2
 
-    qdir_md5 = qdirs_check("quarantine", Path(qdir, "md5-002"), logger)
-    duplicates = qdirs_check("duplicates", Path(qdir, "duplicates-002"), logger)
-
-    # The following directory holds tarballs that are quarantined because
-    # of operational errors on the server. They should be retried after
-    # the problem is fixed: basically, move them back into the reception
-    # area for 002 agents and wait.
-    errors = qdirs_check("errors", Path(qdir, "errors-002"), logger)
+    counts = process_tb(config, logger, receive_dir)
 
-    if qdir_md5 is None or duplicates is None or errors is None:
-        return 1
-
-    counts = process_tb(config, logger, receive_dir, qdir_md5, duplicates, errors)
+    # FIXME: I believe we can just log the status instead of creating report
```

Comment:
That's a good design question. Do we want our 0.69[ish] pass-through compatibility server to be logging Elasticsearch server reports? I suspect we probably do, if only so the entire Pbench system can be diagnosed from the logs, and those reports are likely to be the eventual basis for our permanent audit trail logging ... which should include this pass-through operation.

Reply:
Here we will be only dealing with the tarballs that successfully passed and failed. so, do we really need that?

Reply:
I'm inclined to view the shim as being "just another client".  I think we want to be careful about extending special capabilities to it that we wouldn't proffer to other clients.  (Do we want to allow just any (put-authorized) client to log audit events?  Maybe...maybe not....)

Reply:
> I'm inclined to view the shim as being "just another client". I think we want to be careful about extending special capabilities to it that we wouldn't proffer to other clients. (Do we want to allow just any (put-authorized) client to log audit events? Maybe...maybe not....)

Ah, but put-shim is a part of the 0.69 transition _server_ even though it's a _client_ of the 0.72 server.

So the question is whether we want _our server_ to be logging the passthrough actions? Or rather, _how_ we want it to log the passthrough actions. The tool activities have traditionally been logged in Elasticsearch server-report documents, and some argument could be made for retaining that code instead of removing it. Alternately we could just log to a `Logger` and make someone log in to the passthrough server to find what happened... which we might need to do anyway since the server-report documents don't have much detail.

Reply:
Something tells me that you're teetering on the border between "logging" and "auditing".

If "logging" will suffice for our 0.69 (i.e., the shim) needs, then I would go with that (and do it locally, only) and defer "auditing" to 0.72 or beyond.

Reply:
The whole `server-reports` thing has always teetered on the brink; I've taken to thinking of it as primitive audit logging, albeit at the server tool level rather than at the "resource" level, while we also log locally. Whatever we choose to call it, the question is whether we stop recording an Elasticsearch report document for the action of accepting a tarball dropped by ssh.

Reply:
--> the question is whether we stop recording an Elasticsearch report document for the action of accepting a tarball dropped by ssh.

After giving it a thought and this particular issue I believe we should add the report. I have added it and would love to discuss around it.